File: src/main/java/org/profit/ProfitServer.java
package org.profit;

import org.profit.config.AppContext;
import org.profit.server.ServerManager;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * @author TangYing
 */
public class ProfitServer {

	private static final Logger LOG = LoggerFactory.getLogger(ProfitServer.class);

	private boolean started = false;

	public ProfitServer() {

	}

	public void start() {
		ServerManager serverManager = (ServerManager) AppContext.getBean(AppContext.SERVER_MANAGER);
		try {
			serverManager.startServers();

			// Engine start
			Engine.INSTANCE.start();
			started = true;
		} catch (Exception e) {
			LOG.info("Unable to start servers cleanly: {}", e);
		}
	}

	public void stop() {
		ServerManager serverManager = (ServerManager) AppContext.getBean(AppContext.SERVER_MANAGER);
		try {
			serverManager.stopServers();

			// Engine stop
			Engine.INSTANCE.stop();
			started = false;
			LOG.info("Server stopped.");
		} catch (Exception e) {
			LOG.error("Unable to stop servers cleanly: {}", e);
		}
	}

	public boolean isStopped() {
		return !started;
	}
}

File: src/main/java/org/profit/ProfitServer.kt
package org.profit

import org.profit.config.AppContext
import org.profit.server.ServerManager
import org.profit.server.impl.NettyHttpServer
import org.slf4j.LoggerFactory

class ProfitServer {

    private var started = false

    fun start() {
        val nettyHttpServer = AppContext.getBean(AppContext.HTTP_SERVER) as NettyHttpServer
        try {
            nettyHttpServer.startServer()

            // Engine start
            Engine.INSTANCE.start()
            started = true
            LOG.info("Servers started.")
        } catch (e: Exception) {
            LOG.error("Unable to start servers cleanly: {}", e)
        }
    }

    fun stop() {
        // Engine stop
        Engine.INSTANCE.stop()
        started = false
    }

    companion object {
        private val LOG = LoggerFactory.getLogger(ProfitServer::class.java)
    }
}